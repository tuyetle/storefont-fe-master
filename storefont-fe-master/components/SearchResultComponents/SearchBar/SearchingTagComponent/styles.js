import variables from '~/styles/variables';
import { css } from 'glamor';

export default {
  wrapper: css({
    width: '100%',
    padding: '0.25em 0',
    display: 'flex',
    alignItems: 'center',
    '& .Select': {
      display: 'inline-block',
      width: '100%',
      '& .Select-control': {
        outline: 'none',
        border: 'none',
        boxShadow: 'none',
        '&:hover': {
          boxShadow: 'none',
        },
      },
      '& .Select-menu-outer': {
        zIndex: variables.$zIndexDropdown, left: '-3em',
      },
    },
  }),
};
