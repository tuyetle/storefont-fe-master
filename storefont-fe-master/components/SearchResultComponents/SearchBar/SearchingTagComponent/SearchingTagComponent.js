import React, { PureComponent } from 'react';
import { Creatable } from 'react-select';
import PropTypes from 'prop-types';
import { isFunction } from 'lodash';
import SVGSearch from '~/static/assets/img/ic_search.svg';
import styles from './styles';

class SearchingTagComponent extends PureComponent {
  constructor(props) {
    super(props);

    this.state = { multiValue: props.selectedSuggestion };
    this.onChangeHandler = this.onChangeHandler.bind(this);
    this.onInputChange = this.onInputChange.bind(this);
  }

  onChangeHandler(value) {
    this.setState({ multiValue: value });
    const { fieldApi: { setValue, setTouched } } = this.props;
    if (isFunction(setValue)) {
      setTouched(true);
      setValue(value);
    }
  }

  onInputChange(value) {
    this.props.getSuggestions(value);
  }

  render() {
    const { multiValue } = this.state;
    const {
      suggestions,
      promptText,
      noResultsText,
      placeholder,
      loadingPlaceholder,
    } = this.props;

    return (
      <div className={`${styles.wrapper}`}>
        <SVGSearch />
        <Creatable
          multi
          options={suggestions}
          onChange={this.onChangeHandler}
          onInputChange={this.onInputChange}
          value={multiValue}
          promptTextCreator={label => `${promptText} ${label}`}
          noResultsText={noResultsText}
          placeholder={placeholder}
          loadingPlaceholder={loadingPlaceholder}
          arrowRenderer={() => {}}
        />
      </div>
    );
  }
}

SearchingTagComponent.propTypes = {
  suggestions: PropTypes.array,
  promptText: PropTypes.string,
  noResultsText: PropTypes.string,
  placeholder: PropTypes.string,
  loadingPlaceholder: PropTypes.string,
  getSuggestions: PropTypes.func.isRequired,
  selectedSuggestion: PropTypes.object,

  fieldApi: PropTypes.object.isRequired,
};

SearchingTagComponent.defaultProps = {
  suggestions: [],
  promptText: '',
  noResultsText: '',
  placeholder: '',
  loadingPlaceholder: '',
  selectedSuggestion: undefined,
};

export default SearchingTagComponent;
