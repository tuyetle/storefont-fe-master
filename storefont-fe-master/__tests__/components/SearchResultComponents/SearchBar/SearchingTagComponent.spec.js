import React from 'react';
import renderer from 'react-test-renderer';
import SearchingTagComponent from '~/components/SearchResultComponents/SearchBar/SearchingTagComponent/SearchingTagComponent';

describe('(Component) SearchingTagComponent', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      fieldApi: {},
      getSuggestions: () => {},
    };

    component = <SearchingTagComponent {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

