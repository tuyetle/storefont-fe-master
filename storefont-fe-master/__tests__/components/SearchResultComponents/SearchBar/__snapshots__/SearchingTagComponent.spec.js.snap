// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SearchingTagComponent should render conrectly 1`] = `
<div
  className="css-15sezzo"
>
  <svg
    height="18"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.297 16.282l-4.22-4.388a7.134 7.134 0 0 0 1.68-4.6c0-3.946-3.21-7.157-7.156-7.157S.444 3.347.444 7.294c0 3.946 3.21 7.156 7.157 7.156a7.075 7.075 0 0 0 4.1-1.295l4.251 4.421a.929.929 0 0 0 1.319.025.935.935 0 0 0 .026-1.319zM7.601 2.004a5.296 5.296 0 0 1 5.289 5.29 5.295 5.295 0 0 1-5.289 5.289 5.295 5.295 0 0 1-5.29-5.289 5.296 5.296 0 0 1 5.29-5.29z"
      fill="#606060"
    />
  </svg>
  Creatable
</div>
`;
